plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/BaseDeltaLakeConnectorSmokeTest.java

```diff
@@ -1751,6 +1760,110 @@ public void testOptimizeUsingForcedPartitioning()
         assertThat(getAllDataFilesFromTableDirectory(tableName)).isEqualTo(union(initialFiles, updatedFiles));
     }
 
+    @Test
+    public void testRegisterTableWithTableLocation()
+    {
+        String tableName = "test_register_table_with_table_location_" + randomNameSuffix();
+
+        assertUpdate(format("CREATE TABLE %s (a int, b varchar, c boolean)", tableName));
+        assertUpdate(format("INSERT INTO %s values(1, 'INDIA', true)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s values(2, 'USA', false)", tableName), 1);
+
+        String tableLocation = getTableLocation(tableName);
+        String showCreateTableOld = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+        // Drop table from metastore and use the same table to register again
+        dropTableFromMetastore(tableName);
+
+        assertUpdate("CALL system.register_table (CURRENT_SCHEMA, '" + tableName + "', '" + tableLocation + "')");
+        String showCreateTableNew = (String) computeActual("SHOW CREATE TABLE " + tableName).getOnlyValue();
+
+        assertThat(showCreateTableOld).isEqualTo(showCreateTableNew);
+        assertQuery("SELECT * FROM " + tableName, "VALUES (1, 'INDIA', true),  (2, 'USA', false)");
+
+        assertUpdate("DROP TABLE " + tableName);
+    }
+
+    @Test
+    public void testRegisterTableWithComments()
+    {
+        String tableName = "test_register_table_with_comments_" + randomNameSuffix();
+
+        assertUpdate(format("CREATE TABLE %s (a int, b varchar, c boolean)", tableName));
+        assertUpdate(format("INSERT INTO %s values(1, 'INDIA', true)", tableName), 1);
+        assertUpdate(format("INSERT INTO %s values(2, 'USA', false)", tableName), 1);
```

Comment:
Is there any reason we insert two rows (not one row) here and in other places?

Reply:
There was no specific reason. Updated with one row.